src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_eckey.c

```diff
@@ -0,0 +1,189 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "pal_eckey.h"
+
+#include <assert.h>
+
+EC_KEY* AndroidCryptoNative_NewEcKey(jobject curveParameters, jobject keyPair)
+{
+    assert(curveParameters);
+    assert(keyPair);
+
+    EC_KEY* keyInfo = malloc(sizeof(EC_KEY));
+    memset(keyInfo, 0, sizeof(EC_KEY));
+    atomic_init(&keyInfo->refCount, 1);
+    keyInfo->curveParameters = curveParameters;
+    keyInfo->keyPair = keyPair;
+    return keyInfo;
+}
+
+#pragma clang diagnostic push
+// There's no way to specify explicit memory ordering for increment/decrement with C atomics.
+#pragma clang diagnostic ignored "-Watomic-implicit-seq-cst"
+void AndroidCryptoNative_EcKeyDestroy(EC_KEY* r)
+{
+    if (r)
+    {
+        int count = --r->refCount;
```

Comment:
Should be atomic

Reply:
This is atomic. The type of `refCount` is a C atomic int.